sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/security/authroisation/AuthorisationService.java

```diff
@@ -1,42 +1,74 @@
+
+    private final SecurityContextComponent securityContextComponent;
+
+    private final PractitionerServiceClient practitionerServiceClient;
+    private final RoleServiceClient roleServiceClient;
+
+    @Autowired
+    public AuthorisationService(final SecurityContextComponent securityContextComponent,
+                                final PractitionerServiceClient practitionerServiceClient,
+                                final RoleServiceClient roleServiceClient) {
+        this.securityContextComponent = securityContextComponent;
+        this.practitionerServiceClient = practitionerServiceClient;
+        this.roleServiceClient = roleServiceClient;
+    }
+
     /**
      * Authorise request
      * @return true if authorised
      */
-    @SuppressWarnings("squid:S2589") //allow using method that always return true, currently a stub algorithm
-    public boolean authorise(String requiredPermission) {
-        //currently
-        String userId = "stubUserId";
+    public boolean authorise(String requiredPermission)  {
 
-        //get practitioner assignment role
-        List<String> participantRoles = getParticipantRoles(userId);
+        try {
+            //Get Azure Active Directory user object id
+            String userId = getUserIdFromContext();
 
-        //get permissions for the the practitioner assignment roles
-        List<String> permissions = participantRoles.stream().flatMap(role ->
-                getRolePermissions(role).stream()).collect(Collectors.toList());
+            //get practitioner assignment role
+            AssignmentRoleDTO[] participantRoles = practitionerServiceClient.getUserAssignmentRoles(userId);
 
-        //validate the required permission is present
-        if (!permissions.contains(requiredPermission)) {
-            return false;
-        }
+            if (participantRoles == null || participantRoles.length == 0) {
+                LOGGER.info("User with id {} has no assignment Roles and therefore is unauthorised.", userId);
+                return false;
+            }
 
-        //we will also update the practitioner roles to be only those who have the permissions to perform the action
-        //currently just a stub to show we filtered them out
+            //get permissions for the the practitioner assignment roles
+            //and filter assignment roles to be only those which have the required permission in them
+            var rolesWithPermission = Stream.of(participantRoles)
+                    .map(role -> roleServiceClient.getRolesById(role.getRoleId()))
+                    .filter(roleDto -> hasRoleRequiredPermission(roleDto, requiredPermission))
+                    .collect(Collectors.toList());
 
-        List<String> updatedParticipantRoles = participantRoles.subList(0, 1);
+            //validate the required permission is present
+            if (rolesWithPermission.isEmpty()) {
```

Comment:
Can just do stream...findFirst().isEmpty() instead, don't need to go through all the roles if one has the permission
```suggestion
            var doesNotHavePermission = Stream.of(participantRoles)
                    .map(role -> roleServiceClient.getRolesById(role.getRoleId()))
                    .filter(roleDto -> hasRoleRequiredPermission(roleDto, requiredPermission))
                    .findFirst().isEmpty();

            //validate the required permission is present
            if (doesNotHavePermission) {
                return false;
            }
```